atomisticparsers/utils/parsers.py

```diff
@@ -179,52 +179,53 @@ def parse_md_workflow(self, data: Dict[str, Any]) -> None:
         self.archive.workflow2 = sec_workflow
 
     def parse_interactions(self, interactions: List[Dict], sec_model: MSection) -> None:
-        interaction_dict = {}
-        for interaction_key in Interaction.m_def.all_quantities.keys():
-            interaction_dict[interaction_key] = np.array(
-                [interaction.get(interaction_key) for interaction in interactions],
-                dtype=object,
-            )
-        interaction_dict = {key: val for key, val in interaction_dict.items()}
-        interaction_types = (
-            np.unique(interaction_dict["type"])
-            if interaction_dict.get("type") is not None
-            else []
-        )
-        for interaction_type in interaction_types:
+        if not interactions:
+            return
+
+        def write_interaction_values(values):
             sec_interaction = Interaction()
             sec_model.contributions.append(sec_interaction)
-            interaction_indices = np.where(
-                interaction_dict["type"] == interaction_type
-            )[0]
-            sec_interaction.type = interaction_type
-            sec_interaction.n_interactions = len(interaction_indices)
-            sec_interaction.n_atoms
-            for key, val in interaction_dict.items():
+            sec_interaction.type = current_type
+            sec_interaction.n_atoms = max(
+                [len(v) for v in values.get("atom_indices", [[0]])]
```

Comment:
Should we really give a default of [[0]] if there are no atom_indices instead of just not populating any interactions sections? Can this even occur?

Reply:
Yes sure I can put a return here.